generator/src/generator.rs

```diff
@@ -0,0 +1,767 @@
+
+        let param = MintCis2TokenParam {
+            token_amount: TokenAmount::from(u64::MAX),
+            metadata_url: MetadataUrl::new("https://example.com".into(), None)?,
+        };
+        let params = MintCis2TokenParams {
+            owner:  Address::Account(args.keys.address),
+            tokens: [(TokenId::new_u8(0), param)].into(),
+        };
+
+        let payload = UpdateContractPayload {
+            amount:       Amount::zero(),
+            address:      contract_address,
+            receive_name: OwnedReceiveName::new("cis2_multi.mint".into())?,
+            message:      OwnedParameter::from_serial(&params)?,
+        };
+
+        let mint_tx =
+            args.make_update_contract_transaction(payload, Energy::from(2740), &mut nonce.nonce)?;
+        let block_item = BlockItem::AccountTransaction(mint_tx);
+        let transaction_hash = args.client.send_block_item(&block_item).await?;
+
+        let (_, summary) = args.client.wait_until_finalized(&transaction_hash).await?;
+        anyhow::ensure!(
+            summary.is_success(),
+            "Mint transaction failed (hash = {transaction_hash})."
+        );
+        println!(
+            "Minted u64::MAX tokens (hash = {transaction_hash}, energy = {}).",
+            summary.energy_cost,
+        );
+
+        Ok(Self {
+            args,
+            contract_address,
+            accounts,
+            nonce: nonce.nonce,
+            count: 0,
+        })
+    }
+}
+
+impl Generate for TransferCis2Generator {
+    fn generate(&mut self) -> anyhow::Result<AccountTransaction<EncodedPayload>> {
+        let next_account = self.accounts[self.count % self.accounts.len()];
+        let params = TransferParams::new(
+            [Transfer {
+                token_id: TokenId::new_u8(0),
+                amount:   TokenAmount::from(1u32),
+                from:     Address::Account(self.args.keys.address),
+                to:       Receiver::Account(next_account),
+                data:     AdditionalData::new(vec![])?,
+            }]
+            .to_vec(),
+        )?;
+
+        let payload = UpdateContractPayload {
+            amount:       Amount::zero(),
+            address:      self.contract_address,
+            receive_name: OwnedReceiveName::new("cis2_multi.transfer".into())?,
+            message:      OwnedParameter::from_serial(&params)?,
+        };
+        let energy = Energy::from(3500);
+        let tx = self
+            .args
+            .make_update_contract_transaction(payload, energy, &mut self.nonce)?;
+        self.count += 1;
+
+        Ok(tx)
+    }
+}
+
+/// A generator that makes transactions that wrap, unwrap, and transfer WCCDs.
+pub struct WccdGenerator {
+    args:             CommonArgs,
+    contract_address: ContractAddress,
+    nonce:            Nonce,
+    count:            usize,
+}
+
+#[derive(concordium_std::Serial)]
+struct SetMetadataUrlParams {
+    url:  String,
+    hash: Option<[u8; 32]>,
```

Comment:
There is a type called `metadataUrl` in cis2_types which you should use here instead of defining your own.

Reply:
I get a ParseError when using this, I think because it has a `#[concordium(size_length = 2)]` and the contract doesn't.

Reply:
I see. That's an unfortunate omission in the contract. Oh well, let's leave it then.